javacutil/src/main/java/org/checkerframework/javacutil/AnnotationBuilder.java

```diff
@@ -274,32 +276,39 @@ public AnnotationMirror build() {
    * doesn't exist in the annotation to be built, an error is raised unless the element is specified
    * in {@code ignorableElements}.
    *
-   * @param valueHolder the annotation that holds the values to be copied
-   * @param ignorableElements the elements that can be safely dropped
+   * @param other the annotation that holds the values to be copied; need not be an annotation of
+   *     the same type of the one being build
+   * @param ignorableElements the name of the elements of {@code other} that can be safely dropped
    */
-  public void copyElementValuesFromAnnotation(
-      AnnotationMirror valueHolder, String... ignorableElements) {
-    Set<String> ignorableElementsSet = new HashSet<>(Arrays.asList(ignorableElements));
-    copyElementValuesFromAnnotation(valueHolder, ignorableElementsSet);
+  public void copyElementValuesFromAnnotation(AnnotationMirror other, String... ignorableElements) {
+    for (Map.Entry<? extends ExecutableElement, ? extends AnnotationValue> eltValToCopy :
+        other.getElementValues().entrySet()) {
+      Set<String> ignorableElementsSet = new HashSet<>(Arrays.asList(ignorableElements));
```

Comment:
It seems wasteful to re-create this same set on every iteration through the loop.
It is only used to test containment, so I think you could just do a containment check on the original array or on the `Arrays.asList()` result, without the extra cost of making a set.